histomicstk/segmentation/label/trace_object_boundaries.py

```diff
@@ -117,7 +124,55 @@ def trace_object_boundaries(im_label,
                 im_label, dtype=np.int), conn, x_start, y_start, max_length
         )
 
-        X.append(bx)
-        Y.append(by)
+        bx, by = _remove_thin_colinear_spurs(bx, by, eps_colinear_area)
+
+        if len(bx) > 0:
+            X.append(bx)
+            Y.append(by)
 
     return X, Y
+
+
+def _remove_thin_colinear_spurs(px, py, eps_colinear_area=0.0):
+    """Simplifies the given list of points by removing colinear spurs
+    """
+
+    keep = []  # indices of points to keep
+
+    anchor = -1
+    testpos = 0
+
+    while testpos < len(px):
+
+        # get coords of next triplet of points to test
+        if testpos == len(px) - 1:
+            if not len(keep):
+                break
+            nextpos = keep[0]
+        else:
+            nextpos = testpos + 1
+
+        ind = [anchor, testpos, nextpos]
+        x1, x2, x3 = px[ind]
+        y1, y2, y3 = py[ind]
+
+        # compute area of triangle formed by triplet
+        area = np.linalg.det(
+            np.array([[x1, x2, x3], [y1, y2, y3], [1, 1, 1]])
+        )
```

Comment:
This is twice the triangle's area, so we should either divide it by 2 or rename it and compare it to twice the limiting area.